### PR TITLE
Unit tests for SPF and DMARC

### PR DESCRIPTION
**Pull request by bayerhonza**
While working #254, I wrote few unit tests. I created a new PR as I added a new package for Go unit testing [stretchr/testify](https://github.com/stretchr/testify) and I was not sure if ZDNS maintainers agree with it.

If the addition of such a library is not welcome, I will rewrite the tests so that no other library is needed.

### CODE REVIEW

**Comment by dadrian** [comment]
Would you be willing to switch from testify to `gotest.tools`? It has basically the same API, but is still maintained and has a couple extra features.

**Comment by bayerhonza** [comment]
Ok, done.

**Comment by spencerdrak** [comment]
This looks good to me. If @dadrian doesn't have any other thoughts, I will merge. The checks are passing.

File: modules/bindversion/bindversion.go [hunk 1]
```diff
@@ -34,14 +34,10 @@ type Lookup struct {
...
 func (s *Lookup) DoLookup(_, nameServer string) (interface{}, zdns.Trace, zdns.Status, error) {
-	var res Result
-	res.Resolver = nameServer
-	innerRes, trace, status, err := s.DoTxtLookup("VERSION.BIND", nameServer)
-	if status != zdns.STATUS_NOERROR {
-		return res, trace, status, err
-	}
-	res.BindVersion = innerRes
-	return res, trace, zdns.STATUS_NOERROR, nil
+	innerRes, trace, status, err := s.DoLookup("VERSION.BIND", nameServer)
```
Comment (vishalmohanty) [inline modules/bindversion/bindversion.go]: Isn't this calling the same `DoLookup`? Infinite loop?

**Comment by vishalmohanty** [comment]
Also @bayerhonza can you add UTs for bindversion?

File: modules/bindversion/bindversion.go (hunk 1 above)
Comment (bayerhonza) [reply modules/bindversion/bindversion.go]: Yes it is obviously a bug, I fixed it

**Comment by bayerhonza** [comment]
I've added UTs for bind and improved UTs for SPF and DMARC. Also, I initially created this PR to separate UTs from the implementation itself that is discussed in #254. Shall I close this one and merge everything in #254 or are you going to merge this PR #256 and close the other?

**Comment by spencerdrak** [comment]
Merged 254 which included this, so I think we're all good. Thanks!

**Pull request merged**